legacy/expressions/visitor.go

```diff
@@ -193,50 +178,35 @@ func (v *legacyVisitor) VisitAdditionOrSubtraction(ctx *gen.AdditionOrSubtractio
 		op = "-"
 	}
 
-	secondIsDate := isDate(next)
-	if secondIsDate {
-		secondSeconds, ok := convertTimeToSeconds(next)
-		if ok {
-			next = secondSeconds
-			dateUnit = "s"
-		}
-	}
+	// see if either of our arguments is a date value
+	arg1Type := inferType(arg1)
+	arg2Type := inferType(arg2)
 
-	// see if our second param is an int
-	_, secondNumberErr := strconv.Atoi(next)
-	if (firstIsDate || secondIsDate) && (firstNumberErr != nil || secondNumberErr != nil) {
+	//fmt.Printf("Migrating add/sub with types: %s => %s, %s =>%s\n", arg1, arg1Type, arg2, arg2Type)
 
-		// we are adding two values where we know at least one side is a date
-		template := "datetime_add(%s, %s, \"%s\")"
-		if op == "-" {
-			template = "datetime_add(%s, -%s, \"%s\")"
-		}
+	if arg1Type == "number" && arg2Type == "number" {
+		// we are adding two numbers
+		return fmt.Sprintf("%s %s %s", arg1, op, arg2)
 
-		// determine the order of our parameters
-		replacements := []interface{}{value, next, dateUnit}
```

Comment:
order in legacy expressions has to be date+number so don't have to handle number+date